circle.yml

```diff
@@ -0,0 +1,138 @@
+---
+templates:
+  defaults: &defaults
+    docker:
+      - image: circleci/node:10-browsers
+        environment:
+          JOBS: 2
+
+  node_key: &node_key
+    key: node-deps-{{ checksum "yarn.lock" }}-{{ checksum "circle.yml" }}
+
+  restore_node: &restore_node
+    restore_cache:
+      <<: *node_key
+
+  deploy_static: &deploy_static
+    <<: *defaults
+    steps:
+      - checkout
+      - <<: *restore_node
+      - deploy:
+          name: Deploy to S3
+          command: |
+            env | sed -n "s/^${ENV}_//p" >> .env && source .env
+            nuxt generate
+            gulp deploy
+
+  filter_demo: &filter_demo
+    filters:
+      branches:
+        only: main
+      tags:
+        only: demo
+
+  deploy_nuxt: &deploy_nuxt
+    docker:
+      - image: circleci/python:3.6
+    steps:
+      - checkout
+      - setup_remote_docker
+      - restore_cache:
+          key: deploy-{{ checksum "circle.yml" }}
+      - run:
+          name: Install Requirements
+          command: |
+            if [[ ! -d ~/.venv ]]; then
+              python3 -m venv ~/.venv
+              source ~/.venv/bin/activate
+              pip3 install -U git+https://github.com/nypublicradio/nyprsetuptools.git
+            fi
+      - deploy:
+          name: Deploy Nuxt
+          command: |
+            TAG=${TAG:-$CIRCLE_TAG}
+            source ~/.venv/bin/activate
+            nyprsetuptools DockerDeploy \
+              --fargate \
+              --cpu=1024 \
+              --memory-reservation=2048 \
+              --ecr-repository=wnyc-client \
+              --ecs-cluster-override=fastboot-demo \
+              --execution-role=wnyc-client-$ENV \
+              --task-role=wnyc-client-$ENV \
+              --ports=80 \
+              --environment=$ENV \
+              --tag=$TAG \
+              --wait=300 \
+      - save_cache:
+          key: deploy-{{ checksum "circle.yml" }}
+          paths:
+            - ~/.venv
+
+version: 2
+jobs:
+  build-static:
+    <<: *defaults
+    steps:
+      - checkout
+      - <<: *restore_node
+      - run:
+          name: Install node dependencies
+          command: |
+            if [ ! -d node_modules ]; then
+              yarn --pure-lockfile
+            fi
+      - save_cache:
+          <<: *node_key
+          paths:
+            - node_modules
+
+  test-static:
+    <<: *defaults
+    steps:
+      - checkout
+      - <<: *restore_node
+      - run:
+          name: Test
+          command: npm test
```

Comment:
this step might fail every time because there haven't been tests added yet.

Reply:
we have unit tests - you can run them by doing _npm run test_

Reply:
ah ty, I'll update that

Reply:
@kimlarocca i'm still getting the message saying that there are no tests. Is there some test config file missing?

Reply:
https://github.com/nypublicradio/wnyc-3000-vue/blob/main/test/Logo.spec.js is all commented out